src/main/java/com/minecolonies/coremod/blocks/BlockTimberFrame.java

```diff
@@ -63,104 +52,109 @@ private void initBlock(final String name)
         setHardness(BLOCK_HARDNESS);
         setResistance(RESISTANCE);
     }
-
+    
+    /**
+     * Calc the default state depending on the neighboring blocks.
+     * @deprecated remove this when not needed anymore
+     * @param state
+     * @param world
+     * @param pos
+     * @return
+     */
+    @Deprecated
     @Override
     public IBlockState getActualState(final IBlockState state, final IBlockAccess world, final BlockPos pos)
     {
         final IBlockState upState = world.getBlockState(pos.up());
         final IBlockState downState = world.getBlockState(pos.down());
+
+        final IBlockState leftState = world.getBlockState(pos.east());
+        final IBlockState rightState = world.getBlockState(pos.west());
+
+        final IBlockState straightState = world.getBlockState(pos.south());
+        final IBlockState backState = world.getBlockState(pos.north());
+
         final boolean up = isConnectable(upState);
         final boolean down = isConnectable(downState);
 
-        if(!isConnectable(state) || state.getValue(TYPE) == TimberFrameType.HORIZONTALNOCAP || (!up && !down))
+        final boolean left = isConnectable(leftState);
+        final boolean right = isConnectable(rightState);
+
+        final boolean straight = isConnectable(straightState);
+        final boolean back = isConnectable(backState);
+
+        if(!isConnectable(state) || state.getBlock().getUnlocalizedName().contains(TimberFrameType.HORIZONTALNOCAP.getName())
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (10) used in the expression (maximum allowed 6). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)